pytket/qir/conversion/api.py

```diff
@@ -85,9 +85,30 @@ def pytket_to_qir(
 
     if pyqir_0_7_compatibility:
 
+        if len(circ.c_registers) > 1:
+            raise ValueError(
+                """The qir optimised for pyqir 0.7 can only contain 
+one classical register"""
+            )
+
         initial_result = str(populated_module.module.ir())  # type: ignore
 
-        result = initial_result.replace("entry_point", "EntryPoint").replace("num_required_qubits", "requiredQubits").replace("num_required_results", "requiredResults")  # type: ignore
+        initial_result = initial_result.replace("entry_point", "EntryPoint").replace("num_required_qubits", "requiredQubits").replace("num_required_results", "requiredResults")  # type: ignore
+
+        def keep_line(line: str) -> bool:
+            return (
+                ("@__quantum__qis__read_result__body" not in line)
+                and ("@set_one_bit_in_reg" not in line)
+                and ("@reg2var" not in line)
+                and ("@read_bit_from_reg" not in line)
+                and ("@set_all_bits_in_reg" not in line)
+            )
+
+        result = "\n".join(filter(keep_line, initial_result.split("\n")))
+
+        result = result.replace("i64 %0", "i64 0")
```

Comment:
What is the effect of this? Probably needs a comment at least?

Reply:
It will replace the ssa variable of the register which we are not creating any more with an integer it can use (in this case 0)